streaming/java/streaming-api/src/main/java/org/ray/streaming/api/stream/KeyDataStream.java

```diff
@@ -5,18 +5,29 @@
 import org.ray.streaming.api.partition.impl.KeyPartition;
 import org.ray.streaming.operator.StreamOperator;
 import org.ray.streaming.operator.impl.ReduceOperator;
+import org.ray.streaming.python.stream.PythonDataStream;
+import org.ray.streaming.python.stream.PythonKeyDataStream;
 
 /**
  * Represents a DataStream returned by a key-by operation.
  *
  * @param <K> Type of the key.
  * @param <T> Type of the data.
  */
+@SuppressWarnings("unchecked")
 public class KeyDataStream<K, T> extends DataStream<T> {
 
   public KeyDataStream(DataStream<T> input, StreamOperator streamOperator) {
     super(input, streamOperator);
-    this.partition = new KeyPartition();
+    super.setPartition(new KeyPartition());
```

Comment:
The super class should have a constructor that takes `partition` as parameter.

Reply:
fixed